Add dbt Fusion documentation (#1824)
astronomer/astronomer-cosmos: docs/configuration/dbt-fusion.rst

.. _dbt_fusion:

dbt Fusion support
==================

.. note::
    Available since Cosmos 1.11 when using ``ExecutionMode.LOCAL``.

Context
-------

> dbt Labs `launched <https://www.getdbt.com/blog/dbt-launch-showcase-2025-recap>`_ `dbt Fusion <https://github.com/dbt-labs/dbt-fusion>`_ on 28 May 2025.

Reviewer: ```suggestion
dbt Labs `launched <https://www.getdbt.com/blog/dbt-launch-showcase-2025-recap>`_ `dbt Fusion <https://github.com/dbt-labs/dbt-fusion>`_ on 28 May 2025. dbt Fusion is the next-generation dbt engine that enables real-time model validation, improved SQL parsing and state-aware orchestration. 
```

It is a unified approach that aims to merge dbt Core and dbt Cloud features via a completely new CLI,
implemented in a different programming language (Rust, as opposed to Python).
As part of this, dbt Labs are `rewriting all dbt adaptors <https://github.com/dbt-labs/dbt-fusion/tree/main/crates/dbt-fusion-adapter/src/adapters>`_ (equivalent to Airflow providers) in Rust, starting from Snowflake.

> They are also changing the `licensing model <https://github.com/dbt-labs/dbt-fusion/blob/main/LICENSES.md>`_ to a hybrid Open-Source and commercial license, which has implications for Cosmos.

Reviewer: ```suggestion
They are also changing the `licensing model <https://github.com/dbt-labs/dbt-fusion/blob/main/LICENSES.md>`_ to a hybrid Open-Source and commercial license. We're supporting dbt Fusion with a license-complaint integration. This integration enables teams to:
 - Use dbt Fusion locally for enhanced development experience with real-time validation
 - Deploy dbt Fusion on Astro for production compliance with ELv2 licensing restrictions
 - Maintain consistent workflows across development and production environments
 
```

> Among the advantages of using dbt Fusion is a significant performance improvement (claimed to be 30x, for some models) and AI features.

Reviewer: ```suggestion
Some reported dbt Fusion features include: 
 - **Lightning-fast performance:** Up to 30× faster parsing speeds
 - **Smarter orchestration:** Build only what’s changed, thanks to Fusion’s state-awareness
 - **Real-time dev experience:** Catch errors instantly and explore lineage as you code, if you're using the VS Code extension
 - **Multi-dialect support:** Instant error detection across Snowflake, Databricks, BigQuery, and Redshift SQL dialects

> Note: dbt Fusion is in public beta with current support for Snowflake and Databricks projects, with additional data platforms coming soon. 
```
